elements/lisk-api-client/test/unit/transaction.spec.ts

```diff
@@ -0,0 +1,716 @@
+							type: 'object',
+							properties: {
+								version: { dataType: 'uint32', fieldNumber: 1 },
+								timestamp: { dataType: 'uint32', fieldNumber: 2 },
+								height: { dataType: 'uint32', fieldNumber: 3 },
+								previousBlockID: { dataType: 'bytes', fieldNumber: 4 },
+								transactionRoot: { dataType: 'bytes', fieldNumber: 5 },
+								generatorPublicKey: { dataType: 'bytes', fieldNumber: 6 },
+								reward: { dataType: 'uint64', fieldNumber: 7 },
+								asset: {
+									type: 'object',
+									fieldNumber: 8,
+									properties: {
+										maxHeightPreviouslyForged: { dataType: 'uint32', fieldNumber: 1 },
+										maxHeightPrevoted: { dataType: 'uint32', fieldNumber: 2 },
+										seedReveal: { dataType: 'bytes', fieldNumber: 3 },
+									},
+									required: ['maxHeightPreviouslyForged', 'maxHeightPrevoted', 'seedReveal'],
+								},
+								signature: { dataType: 'bytes', fieldNumber: 9 },
+							},
+							required: [
+								'version',
+								'timestamp',
+								'height',
+								'previousBlockID',
+								'transactionRoot',
+								'generatorPublicKey',
+								'reward',
+								'asset',
+							],
+							fieldNumber: 2,
+						},
+					},
+				},
+			},
+			{
+				moduleID: 1000,
+				moduleName: 'legacyAccount',
+				assetID: 0,
+				assetName: 'reclaimLSK',
+				schema: {
+					$id: 'lisk/legacyAccount/reclaim',
+					title: 'Reclaim transaction asset',
+					type: 'object',
+					required: ['amount'],
+					properties: { amount: { dataType: 'uint64', fieldNumber: 1 } },
+				},
+			},
+		],
+	};
+
+	beforeEach(() => {
+		channelMock = {
+			connect: jest.fn(),
+			disconnect: jest.fn(),
+			invoke: jest.fn(),
+			subscribe: jest.fn(),
+		};
+
+		when(channelMock.invoke)
+			.calledWith('app:getAccount')
+			.mockResolvedValue(accountHex as never)
+			.calledWith('app:getTransactionByID')
+			.mockResolvedValue(txHex as never)
+			.calledWith('app:getTransactionsFromPool')
+			.mockResolvedValue([txHex] as never);
+
+		transaction = new Transaction(channelMock, schema, nodeInfo);
+	});
+
+	describe('Transaction', () => {
+		describe('constructor', () => {
+			it('should initialize with channel', () => {
+				expect(transaction['_channel']).toBe(channelMock);
+			});
+		});
+
+		describe('get', () => {
+			it('should invoke app:getTransactionByID', async () => {
+				// Act
+				await transaction.get(txId);
+
+				// Assert
+				expect(channelMock.invoke).toHaveBeenCalledTimes(1);
+				expect(channelMock.invoke).toHaveBeenCalledWith('app:getTransactionByID', {
+					id: txId.toString('hex'),
+				});
+			});
+		});
+
+		describe('getFromPool', () => {
+			it('should invoke app:getTransactionsFromPool', async () => {
+				// Act
+				await transaction.getFromPool();
+
+				// Assert
+				expect(channelMock.invoke).toHaveBeenCalledTimes(1);
+				expect(channelMock.invoke).toHaveBeenCalledWith('app:getTransactionsFromPool');
+			});
+		});
+
+		describe('create', () => {
```

Comment:
`create` and `sign` has a lot of condtions in the logic.
I think each of those cases should be tested.
In general we do like
```
describe('create',  ...
  describe('when XXX happens',
    it('should be ...
  describe('when YYY happens',
    it('should be ...
```